C-Sharp/dp/BinarySearchTree.cs

```diff
@@ -0,0 +1,233 @@
+        public int FindMax()
+        {
+            return FindMax(root);
+        }
+        //To find the max value in BST
+        public int FindMax(Node root)
+        {
+            if (root.right is null)
+            {
+                return root.data;
+            }
+            return FindMax(root.right);
+        }
+        //To search for a specific node in BST
+        public Node Find(int data)
+        {
+            Node current = root;
+            while (data != current.data)
+            {
+                if (data < current.data)
+                {
+                    current = current.left;
+                }
+                else
+                {
+                    current = current.right;
+                }
+                if (current is null)
+                {
+                    Console.WriteLine("Node is not found.");
+                    return null;
+                }
+            }
+            return current;
+        }
+        //To limit number of parameters
+        public void Delete(int data)
+        {
+            root = Delete(root, data);
+        }
+        //To Delete a specific node
+        public Node Delete(Node root, int data)
+        {
+            if (root == null)
+                return root;
+            //search for the node with the needed value
+            if (data < root.data)
+                root.left = Delete(root.left, data);
+            else if (data > root.data)
+                root.right = Delete(root.right, data);
+            else
+            {
+                //for nodes with one or none child
+                if (root.left == null)
+                    return root.right;
+                else if (root.right == null)
+                    return root.left;
+                //for nodes with two childs get min value in right
+                root.data = FindMin(root.right);
+                root.right = Delete(root.right, root.data);
+            }
+            return root;
+        }
+        public static void Main(String[] args)
+        {
+            BinarySearchTree BST = new BinarySearchTree();
+            BST.Insert(50);
+            BST.Insert(30);
+            BST.Insert(20);
+            BST.Insert(40);
+            BST.Insert(70);
+            BST.Insert(60);
+            BST.Insert(80);
+
+            Console.WriteLine("Inorder traversal");
+            BST.Inorder();
+
+            Console.WriteLine("Delete 20");
+            BST.Delete(20);
+            Console.WriteLine("Inorder traversal");
+            BST.Inorder();
+            Console.WriteLine("Preorder traversal");
+            BST.Preorder();
+            Console.WriteLine("Postorder traversal");
+            BST.Postorder();
+            Console.WriteLine("Min Value:");
+            Console.WriteLine(BST.FindMin());
+            Console.WriteLine("Max Value:");
+            Console.WriteLine(BST.FindMax());
+
+
+
+        }
+    }
+}
+/*
+Sample Input: 50 30 20 4070 60 80
```

Comment:
Can you mention the complexity of the code?

Reply:
I would but the pr already closed.